Add mozc, a Japanese input method from Google.

gebner opened the pull request:
Ibus integration works for me with:

```
  programs.ibus.enable = true;
  programs.ibus.plugins = [ pkgs.mozc ];
```

Upstream mozc is composed of several source repositories and uses `gclient` to manage them, there seem to be no real source releases at the moment; so I used the source distribution from debian instead.  We could also fetch the source repositories directly, see for example the [archlinux PKGBUILD](https://aur.archlinux.org/packages/mo/mozc/PKGBUILD).

The installation script is also taken from the PKGBUILD.  Again, there does not seem to be any official installation method.

gebner committed:
Add mozc and its dependency zinnia.

abbradar commented:
It would be good IMO to make a common configuration namespace for input methods: right now we have `config.uim` and `config.programs.ibus`. I suggest `i18n.ime`; willing to make a patch if it's a good idea.

gebner commented:
Since mozc has now moved to github, I've changed the derivation to build directly from github, without using the debian source packages--as far as I can tell most bundled dependencies are not necessary for a desktop build.  However they now [depend on clang](https://code.google.com/p/mozc/issues/detail?id=276).

Unfortunately, there are no releases made from the git repository yet, so this is a build of the current git revision.

gebner commented:
Also changed zinnia to use `fetchFromGitHub` instead of `fetchgit`.

gebner commented:
Changed the mozc derivation to use clangStdenv instead of using clang as a build input.

abbradar commented:
That would be last remarks, I think (sorry for not writing them all in one go).

gebner commented:
Ok, thanks for the review!  I hope I got all the dependencies for `nativeBuildInputs`--is there any way I can test that?

abbradar commented:
Hm, there's a way to understand you've moved too much to nativeBuildInputs (it'll fail to compile), but I don't know if there's way to find out if you have not put everything needed -- worth asking in ML.

That's not much of an issue however, so that aside I'll test build and merge if everything is good later today. Thank you for your work and sorry for you having to wait a long time -- we are somewhat understaffed ^_^;
-- 
Nikolay.
